flower/utils/pg_storage.py

```diff
@@ -0,0 +1,91 @@
+import json
+import logging
+from datetime import datetime
+
+import pg8000
+
+logger = logging.getLogger(__name__)
+connection = None
+skip_callback = False
+
+_all_tables = """
+SELECT * FROM information_schema.tables
+WHERE table_schema = 'public'
+"""
+
+_schema = (
+    """CREATE TABLE events
+    (
+        id SERIAL PRIMARY KEY,
+        time TIMESTAMP,
+        data JSONB NOT NULL
+    )""",
+    "CREATE INDEX event_index ON events USING GIN (data)",
+    "CREATE INDEX event_time_index ON events (time ASC)",
+)
+
+_add_event = """INSERT INTO events (time, data) VALUES (%s, %s)"""
```

Comment:
```
_add_event = """INSERT INTO events (time, data) VALUES (%s, %s) ON CONFLICT DO NOTHING"""
```
postgresql 9.5 supports ON CONFLICT resolution (for upsert, but also good for avoiding dups)